### PR TITLE
[NTP Search]: Serve bundled icons, where possible

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/38503

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue

### CODE REVIEW

File: ui/webui/resources/BUILD.gn [hunk 1]
```diff
@@ -305,6 +305,7 @@ leo_icons = [
   "protocol-handler-off.svg",
   "protocol-handler.svg",
   "qr-code.svg",
+  "qwant-color.svg",
```
Comment (fmarier) [inline ui/webui/resources/BUILD.gn:308]: Did you forget to add these to the PR?

File: components/brave_new_tab_ui/components/search/SearchEngineIcon.tsx [hunk 2]
```diff
@@ -7,9 +7,25 @@ import * as React from "react";
 import { useUnpaddedImageUrl } from "../../../brave_news/browser/resources/shared/useUnpaddedImageUrl";
 import { icon } from "@brave/leo/tokens/css/variables";
 import styled from "styled-components";
+import { SearchEngineInfo } from "../../api/background";
+import Icon from "@brave/leo/react/icon";
+
+// Mapping of search engine host to Nala icon for the icons we bundle in the browser
+const icons = {
+  // The Google search provider has the empty origin :O
+  "": 'google-color',
```
Comment (simonhong) [inline components/brave_new_tab_ui/components/search/SearchEngineIcon.tsx:16]: nit: 
```suggestion
  '': 'google-color',
```

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: ui/webui/resources/BUILD.gn (hunk 1 above)
Comment (fallaciousreasoning) [reply ui/webui/resources/BUILD.gn:308]: Nah, they're already part of Nala :smile:

Comment (fallaciousreasoning) [reply ui/webui/resources/BUILD.gn:308]: https://nala.bravesoftware.com/?path=/story/components-icon--all-icons

**Comment by fmarier** [comment]
If you create a PR to remove https://github.com/brave/brave-core/blob/62a6b2e9323476f495c6be39b90bbf95f68df358/browser/net/brave_network_audit_allowed_lists.h#L74, and then [run the NetworkAuditTest](https://bravesoftware.slack.com/archives/C7VLGSR55/p1716567465760919?thread_ts=1716567098.787939&cid=C7VLGSR55) on it, that would confirm that the image is no longer loaded remotely. All that QA would need to verify for this PR is that the favicon still appears.

I guess we should have done that as part of the original PR, sorry for not catching that.

**Pull request merged**